spec/main_spec.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+require_relative '../lib/game.rb'
+require_relative '../lib/players.rb'
+
+describe Game do
+  let(:first_game) { Game.new }
+  let(:empty_board) do
+    [['', '', ''], ['', '', ''], ['', '', ''], ['', '', ''], ['', '', ''], ['', '', ''], ['', '', ''], ['', '', '']]
+  end
+
+  context 'test initialize values of object with class first_game' do
+    it 'starts turn with X' do
+      expect(first_game.turn).to eq(:X)
+    end
+
+    it 'starts counter with 1' do
+      expect(first_game.counter).to eq(1)
+    end
+
+    it "starts 'result' with empty strings" do
+      expect(first_game.result).to eq('')
+    end
+
+    it 'class of created object is equal to Game' do
+      expect(first_game.class).to eq(Game)
+    end
+  end
+
+  describe '#check_valid_choice?' do
+    it 'should return different to true when no valid choice is given' do
+      expect(first_game.check_valid_choice?('aisdad')).not_to be true
+    end
+
+    it 'should return true when valid choice is given' do
+      expect(first_game.check_valid_choice?('a2')).to be true
+    end
+  end
+
+  describe '#mark_choice' do
+    it 'raise error when no parameter is given' do
+      expect { first_game.mark_choice }.to raise_error(ArgumentError)
+    end
+
+    it 'add choice to @first_game when there is a valid choice' do
+      first_game.mark_choice('b2', 2)
+      expect(first_game.board[:b2]).to eq(:X)
+    end
+
+    it "should increment 'contador' by one when a valid choice is given" do
+      first_game.mark_choice('b2', first_game.contador)
+      expect(first_game.contador).to eq(1)
+    end
```

Comment:
This is a valid case, you need to test for an invalid case also for this one.

Reply:
i could not figure out how to  make an invalid case for this, because #mark_choice only runs after a valid choice is given, and automatically increments 'contador'.

i added another test in 'check_valid_choice' to verify that 'contador' doesnt get modified after an invalid choice

thanks!